R/surv.b.R

```diff
@@ -40,150 +52,237 @@ survClass <- R6::R6Class(
+                DF$pvalue <- pchisq(DF$chisqr, df, lower.tail = FALSE)
                 
-                if (length(groups) >= 2) {
-                    groupsData <- list()
-                    for (group in groups) {
-                        ss <- (data$group == group)
-                        groupsData[[group]] <- subset(data, subset=ss, select=c('elapsed', 'event'))
-                    }
-                }
                 
-                for (pair in tt$rowKeys) {
-                    
-                    x <- groupsData[[pair[1]]]
-                    y <- groupsData[[pair[2]]]
+                #################################
+                ### Populates the table with the  
+                ### tests results
+                #################################
+                tt <- self$results$tests
+                if (tt$isNotFilled() && length(self$options$tests) > 0) {
                     
                     for (i in seq_along(self$options$tests)) {
+                        
                         test <- self$options$tests[i]
                         
-                        if (tt$isFilled(rowKey=pair, col=paste0('nu[', test, ']')))
+                        if (tt$isFilled(rowKey=1, col=paste0('chisqr[', test, ']')))
                             next()
                         
                         private$.checkpoint()
-                        
-                        result <- EnvStats::twoSampleLinearRankTestCensored(
-                            test = test,
-                            x = x$elapsed,
-                            x.censored = ! x$event,
-                            y = y$elapsed,
-                            y.censored = ! y$event,
-                            censoring.side = 'right')
-                        
                         row <- list()
+                        row[[paste0('chisqr[', test, ']')]] <- DF$chisqr[i]
+                        row[[paste0('df[', test, ']')]] <- DF$df[i]
+                        row[[paste0('pvalue[', test, ']')]] <- DF$pvalue[i]
                         
-                        row[[paste0('z[', test, ']')]] <- result$statistic['z']
-                        row[[paste0('nu[', test, ']')]] <- result$statistic['nu']
-                        row[[paste0('nuse[', test, ']')]] <- sqrt(result$statistic['var.nu'])
-                        row[[paste0('p[', test, ']')]] <- result$p.value
-                        
-                        tt$setRow(rowKey=pair, values=row)
+                        tt$setRow(rowKey=1, values=row)
                     }
+                    
                 }
+                tt$setNote("note1","Log-Rank and Peto-Peto tests use survdiff from the survival package. Gehan and Tarone-Ware tests are obtained from the coin package.",init=TRUE)
+
             }
+
+            
+            
+            
+
+            
         },
-        .plot=function(image, theme, ggtheme, ...) {
+        .plot=function(image, ...) {
```

Comment:
oh, can we honour the theme the user has selected?

at the very least we should set the background colour transparent.

Reply:
So that's what it is for?  Sorry I didn't know why it was there. I've put it back.

Reply:
depending on the colour scheme selected from the kebab menu, `theme` and `ggtheme` will contain the different theme/colour scheme information.

you typically just go:

```
plot <- plot + ggtheme
```

and your plots will change colour scheme, to match what the user wants.

Reply:
I've updated this. I had some trouble because ggtheme overrides the legend options (I remove the legend if there is no grouping variable). Luckily this can be included in the ggsurvplot call :-)
So this is fixed now and it looks great !